tests/unit_tests/test_material.py

```diff
@@ -58,6 +58,61 @@ def test_elements_by_name():
     assert a._nuclides == b._nuclides
     assert b._nuclides == c._nuclides
 
+def test_adding_elements_by_formula():
+    """Test adding elements from a formula"""
+    # testing the correct nuclides are added to the Material
+    m = openmc.Material()
+    m.add_elements_from_formula('Li4SiO4')
+    ref_dens = {'Li6': 0.033728, 'Li7': 0.410715,
+                'Si28': 0.102477, 'Si29': 0.0052035, 'Si30': 0.0034301,
+                'O16': 0.443386, 'O17': 0.000168}
+    nuc_dens = m.get_nuclide_atom_densities()
+    for nuclide in ref_dens:
+        assert nuc_dens[nuclide][1] == pytest.approx(ref_dens[nuclide], 1e-2)
+
+    # testing the correct nuclides are added to the Material when enriched
+    m = openmc.Material()
+    m.add_elements_from_formula('Li4SiO4',
+                                enrichment=60.,
+                                enrichment_target='Li6')
+    ref_dens = {'Li6': 0.2666, 'Li7': 0.1777,
```

Comment:
Would it make sense to check the elemental fractions before diving in to the atom densities? This way we can easily verify that the method is assigning four parts lithium to one part silicon to four parts oxygen, independent of the enrichments and abundances?

Reply:
This would be nice and I am sure I can do it. 

I shall see if I can find an elegant way to do it using the existing methods. The .get_nuclide_atom_densities() method makes it easy to check the nuclides but if I have understood correctly there is not an equivalent method for element checking.

Perhaps in the future I can add a few more methods to the material class like get_elements(), get_element_densities(), get_element_atom_densities()

Reply:
As an interim step, you could do something like
```python
elem = collections.defaultdict(float)
for nuclide, adens in m.get_nuclide_atom_densities().values():
    if nuclide.startswith("Li"):
        elem["Li"] += adens
    # repeat for other elements
elem_percents = {key: value / sum(elem.values) for key, value in elem.items())
assert elem_percents["Li"] == pytest.approx(4)
# repeat for other elements
```
Granted, this works only if you know the elements that should exist. Not too much a burden for this test, and doesn't require adding (and testing) new ``Material`` methods

Reply:
That looks great, thanks for the help. I have implement this test